lambdas/drivingpermitcheck/src/main/java/uk/gov/di/ipv/cri/drivingpermit/api/gateway/ThirdPartyDocumentGateway.java

```diff
@@ -207,38 +202,59 @@ private DocumentCheckResult responseHandler(CloseableHttpResponse httpResponse)
             throws IOException, ParseException, JOSEException, CertificateException,
                     OAuthHttpResponseExceptionWithErrorBody {
         int statusCode = httpResponse.getStatusLine().getStatusCode();
-        LOGGER.info("Third party response code {}", statusCode);
 
         HttpEntity entity = httpResponse.getEntity();
         String responseBody = EntityUtils.toString(entity);
-        LOGGER.info("Third party response {}", responseBody);
 
         if (statusCode == 200) {
-            DcsResponse unwrappedDcsResponse =
-                    dcsCryptographyService.unwrapDcsResponse(responseBody);
-            validateDcsResponse(unwrappedDcsResponse);
+            LOGGER.info("Third party response code {}", statusCode);
+
+            try {
+                DcsResponse unwrappedDcsResponse =
+                        dcsCryptographyService.unwrapDcsResponse(responseBody);
+                validateDcsResponse(unwrappedDcsResponse);
+
+                DocumentCheckResult documentCheckResult = new DocumentCheckResult();
+                documentCheckResult.setExecutedSuccessfully(true);
+                documentCheckResult.setTransactionId(unwrappedDcsResponse.getRequestId());
+                documentCheckResult.setValid(unwrappedDcsResponse.isValid());
+
+                return documentCheckResult;
+            } catch (IpvCryptoException e) {
+                // Seen when a signing cert has expired and all message signatures fail verification
+                // We need to log this specific error message from the IpvCryptoException for
+                // context
+                LOGGER.error(e.getMessage());
+                throw new OAuthHttpResponseExceptionWithErrorBody(
+                        HttpStatusCode.INTERNAL_SERVER_ERROR,
+                        ErrorResponse.FAILED_TO_UNWRAP_DCS_RESPONSE);
+            }
+        } else {
 
-            DocumentCheckResult documentCheckResult = new DocumentCheckResult();
-            documentCheckResult.setExecutedSuccessfully(true);
-            documentCheckResult.setTransactionId(unwrappedDcsResponse.getRequestId());
-            documentCheckResult.setValid(unwrappedDcsResponse.isValid());
+            String responseText = responseBody == null ? "No Text Found" : responseBody;
 
-            return documentCheckResult;
-        } else {
-            DocumentCheckResult documentCheckResult = new DocumentCheckResult();
-            documentCheckResult.setExecutedSuccessfully(false);
+            LOGGER.error(
+                    "DCS replied with HTTP status code {}, response text: {}",
```

Comment:
Does this ever contain PII?

Reply:
If the responseBody is successfully mapped to responseText this will definitely include PII

Reply:
When there is non 200 response the response body is an error message.
I have disabled using the text as an error message. 
But I recommend we get confirmation from DCS that only the 200 responses contain information we cannot log. The season being is that the error messages are very informative, normally specify exactly why something has gone wrong or was rejected.